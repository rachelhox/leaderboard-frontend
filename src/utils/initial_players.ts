/* eslint-disable no-undef */
export const initialPlayers: Player[] = [
  {
    name: 'Angel',
    scores: 35,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Angel`,
  },
  {
    name: 'Angus',
    scores: 98,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Angus`,
  },
  {
    name: 'Berry',
    scores: 8,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Berry`,
  },
  {
    name: 'Henry',
    scores: 98,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Henry`,
  },
  {
    name: 'Louis',
    scores: 56,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Louis`,
  },
  {
    name: 'Gus',
    scores: 29,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Gus`,
  },
  {
    name: 'Hannah',
    scores: 79,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Hannah`,
  },
  {
    name: 'Manny',
    scores: 59,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Manny`,
  },
  {
    name: 'Catherine',
    scores: 36,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Catherine`,
  },
  {
    name: 'Audrey',
    scores: 48,
    image: `${process.env.NEXT_PUBLIC_AVATAR_URL}/Audrey`,
  },
];
