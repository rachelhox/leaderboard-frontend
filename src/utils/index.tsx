import { initialPlayers } from './initial_players';
import { getComparator, stableSort } from './sort';

export { initialPlayers, getComparator, stableSort };
