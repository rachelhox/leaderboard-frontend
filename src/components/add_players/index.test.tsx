import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import AddPlayer from '.';

// ==================================
// unit tests
// ==================================
describe('component: AddPlayer', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <AddPlayer />
      </MockTheme>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
