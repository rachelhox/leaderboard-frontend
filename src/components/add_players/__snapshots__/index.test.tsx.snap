// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: AddPlayer matches snapshot 1`] = `
<div
  className="MuiBox-root css-15bwj8y"
>
  <form
    noValidate={true}
    onSubmit={[Function]}
    style={
      {
        "height": "100%",
      }
    }
  >
    <div
      className="MuiBox-root css-iettro"
    >
      <div
        className="MuiBox-root css-v9p2so"
      >
        <button
          aria-label="generate avatar"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AccountCircleSharpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiBox-root css-1uzncql"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wis8bz-MuiTypography-root"
        >
          Name
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root css-zy6lx1-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1jpwi7z-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-describedby="demo-helper-text-aligned-no-helper-helper-text"
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="demo-helper-text-aligned-no-helper"
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Name"
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1dd6dum-MuiFormHelperText-root"
            id="demo-helper-text-aligned-no-helper-helper-text"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
      </div>
      <div
        className="MuiBox-root css-1uzncql"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wis8bz-MuiTypography-root"
        >
          Scores
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root css-zy6lx1-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1jpwi7z-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-describedby="demo-helper-text-aligned-no-helper-helper-text"
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="demo-helper-text-aligned-no-helper"
              name="scores"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Scores"
              required={false}
              type="number"
              value={0}
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1dd6dum-MuiFormHelperText-root"
            id="demo-helper-text-aligned-no-helper-helper-text"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
      </div>
      <button
        aria-label="Add More Player"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-default  css-5zvidf-MuiButtonBase-root-MuiFab-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-yz01is-MuiButtonBase-root-MuiButton-root-AddPlayer"
      disabled={true}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      Add Player(s)
    </button>
  </form>
</div>
`;
